dbgpt/storage/knowledge_graph/graph_retriever/graph_retriever.py

```diff
@@ -0,0 +1,151 @@
+"""Graph Retriever."""
+
+import logging
+from abc import ABC, abstractmethod
+from typing import List, Optional, Union
+
+from pydantic import Field
+
+from dbgpt._private.pydantic import ConfigDict
+from dbgpt.core import Chunk
+from dbgpt.rag.index.base import IndexStoreBase, IndexStoreConfig
+from dbgpt.rag.transformer.text_embedder import TextEmbedder
+from dbgpt.storage.graph_store.graph import (
+    Direction,
+    Edge,
+    Graph,
+    GraphElemType,
+    MemoryGraph,
+    Vertex,
+)
+from dbgpt.storage.knowledge_graph.graph_retriever.base import GraphRetrieverBase
+from dbgpt.storage.knowledge_graph.graph_retriever.document_graph_retriever import (
+    DocumentGraphRetriever,
+)
+from dbgpt.storage.knowledge_graph.graph_retriever.keyword_based_graph_retriever import (
+    KeywordBasedGraphRetriever,
+)
+from dbgpt.storage.knowledge_graph.graph_retriever.text_based_graph_retriever import (
+    TextBasedGraphRetriever,
+)
+from dbgpt.storage.knowledge_graph.graph_retriever.vector_based_graph_retriever import (
+    VectorBasedGraphRetriever,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class GraphRetriever(GraphRetrieverBase):
+    """Graph retriever class."""
+
+    def __init__(
+        self,
```

Comment:
Too many parameters. Configuration is passed through config. Only necessary parameters are retained: `graph_store_apdater`.